### ISSUE TITLE
Move @mui/internal-test-utils to devdepedencies. 

### ISSUE THREAD

**Issue by Matt-Butler** [mui/mui-x#13317]
### Search keywords

@mui/internal-test-utils

### Latest version

- [X] I have tested the latest version

### Steps to reproduce

[Link to live example: (required)

Steps:
](https://github.com/mui/mui-x/blob/a4c6f4b8502923a453e2ccfce5361a0ab1b62a35/packages/x-data-grid/package.json#L51)

### Current behavior

Installing this package, requires updating the version of my test utils. 

### Expected behavior

Installing this package has no impact on the version of my test utils.

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by Matt-Butler** [comment]
https://github.com/search?q=repo%3Amui%2Fmui-x+%40mui%2Finternal-test-utils+language%3AJSON&type=code&l=JSON

**Comment by LukasTy** [comment]
Thank you for reporting it! 🙏 
I'm very sorry that this mistake has slipped through the code review, it's only the `@mui/x-data-grid` package, which has this dependency listed in the regular `dependencies` list rather than `devDependencies`. 🙈

**Comment by Matt-Butler** [comment]
Thanks for the quick response, I appreciate it!

**Comment by LukasTy** [comment]
@Matt-Butler is it a blocking problem for you or waiting for next week's (next Thursday) regular release is a viable option? 🤔

**Comment by LukasTy** [comment]
@Matt-Butler A fixed package (v7.6.1) has been released: https://www.npmjs.com/package/@mui/x-data-grid?activeTab=versions

**Comment by Matt-Butler** [comment]
Thank you so much!
